src/wallet/rpcwallet.cpp

```diff
@@ -1720,6 +1739,47 @@ UniValue walletpassphrasechange(const UniValue& params, bool fHelp)
     return NullUniValue;
 }
 
+UniValue walletconfirmbackup(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() != 1)
+        throw runtime_error(
+            "walletconfirmbackup \"emergency recovery phrase\"\n"
+            "\nNotify the wallet that the user has backed up the emergency recovery phrase,\n"
+            "which can be obtained by making a call to z_exportwallet. The zcashd embedded wallet\n"
+            "requires confirmation that the emergency recovery phrase has been backed up before it\n"
+            "will permit new spending keys or addresses to be generated.\n"
+            "\nArguments:\n"
+            "1. \"emergency recovery phrase\" (string, required) The full recovery phrase returned as part\n"
+            "   of the data returned by z_exportwallet. An error will be returned if the value provided\n"
+            "   does not match the wallet's existing emergency recovery phrase.\n"
+            "\nExamples:\n"
+            + HelpExampleCli("walletconfirmbackup", "\"abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon art\"")
+        );
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    EnsureWalletIsUnlocked();
+
+    SecureString strMnemonicPhrase(params[0].get_str());
+    boost::erase_all(strMnemonicPhrase, "\"");
```

Comment:
Hmm, we should only be erasing quotations from the beginning and end (if they exist, which I'm slightly surprised they would - I'd have thought this would be handled by the JSON-RPC parser). It shouldn't be possible to submit a phrase that has random interleaved quotation marks, as that sends the wrong message to the user (that the phrase with quotations interleaved is a valid mnemonic phrase).

Reply:
The relationship between the arguments passed to zcash-cli and the actual JSON that gets sent to the RPC interface is rather unclear to me. I agree with Larry's assessment though - it's the Python test code that's incorrect.